### PR TITLE
useField docs

### PR DESCRIPTION
**Pull request by devongovett**
Do we also want to document help text support built into other hooks like useSelect/useTextField?

### CODE REVIEW

File: packages/@react-aria/label/docs/useField.mdx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<!-- Copyright 2020 Adobe. All rights reserved.
```
Comment (ktabors) [inline packages/@react-aria/label/docs/useField.mdx:1]: 2021

File: packages/@react-aria/label/docs/useField.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+<ContactPicker
+  label="Preferred contact method"
+  description="Select the best way to contact you about issues with your account." />
+
+<ContactPicker
+  label="Preferred contact method"
+  errorMessage="Select a contact method." />
+```
+
+By default, `useField` assumes that the label is a native HTML `<label>` element. However, if you are labeling a non-native form element, be sure to use an element other than a `<label>` and set the `labelElementType` prop appropriately.
```
Comment (ktabors) [inline packages/@react-aria/label/docs/useField.mdx:77]: Should this paragraph be above the "note"? I feel it is getting lost here.

File: packages/@react-aria/label/docs/useField.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## API
+
+<FunctionAPI function={docs.exports.useField} links={docs.links} />
+
+## Example
+
+The `useField` hook associates a form control with a label, and an optional description and/or error message. This is useful for providing context about how users should fill in a field, or a validation message. `useField` takes care of creating ids for each element and associating them with the correct ARIA attributes (`aria-labelledby` and `aria-describedby`).
+
+**Note**: Many other React Aria hooks such as [useTextField](useTextField.html), [useSelect](useSelect.html), and [useComboBox](useComboBox.html) already include support for description and error message elements. If you're using one of those hooks, there's no need to use `useField`.
```
Comment (ktabors) [inline packages/@react-aria/label/docs/useField.mdx:43]: I'm prefer we didn't say "Note", but I realize if we don't do this and more problems will happen.

**Review by ktabors — DISMISSED** [review]

File: packages/@react-aria/label/docs/useField.mdx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        <option>Email</option>
+        <option>Phone</option>
+        <option>Fax</option>
+        <option>Carrier pigeon</option>
+      </select>
+      {props.description &&
+        <div {...descriptionProps} style={{fontSize: 12}}>{props.description}</div>
+      }
+      {props.errorMessage &&
+        <div {...descriptionProps} style={{color: 'red', fontSize: 12}}>{props.errorMessage}</div>
```
Comment (LFDanLu) [inline packages/@react-aria/label/docs/useField.mdx:62]:
```suggestion
        <div {...errorMessageProps} style={{color: 'red', fontSize: 12}}>{props.errorMessage}</div>
```

**Review by LFDanLu — APPROVED** [review]
LGTM. What did you have in mind for useSearchField/useComboBox/etc? Should they link back to this page and mention that they use useField internally?

**Review by reidbarber — DISMISSED** [review]

**Review by dannify — APPROVED** [review]

File: packages/@react-aria/combobox/docs/useComboBox.mdx [hunk 5]
```diff
@@ -134,7 +139,7 @@ See [useOverlayTrigger](useOverlayTrigger.html) for an example of how to impleme
 using <TypeLink links={overlaysDocs.links} type={overlaysDocs.exports.useOverlayPosition} />.
 
 In addition, see [useListBox](useListBox.html) for examples of sections (option groups), and more complex
-options.
+options, and [useTextField](useTextField.html) for an example of the description and error message elements.
```
Comment (LFDanLu) [inline packages/@react-aria/combobox/docs/useComboBox.mdx:142]: Nit: Reads a bit like `useTextField` is a part of the things within `useListBox`. Perhaps something like this?
```suggestion
options. For an example of the description and error message elements, see [useTextField](useTextField.html).
```

or 
```
In addition, see [useListBox](useListBox.html) for examples of sections (option groups) and more complex options, and [useTextField](useTextField.html) for an example of the description and error message elements.
```
Same for the other instances of this sentence in the other pages

**Review by LFDanLu — APPROVED** [review]
LGTM

**Pull request merged**